horse_racing.rb

```diff
@@ -1,16 +1,21 @@
 # Solution for https://www.codingame.com/ide/puzzle/horse-racing-duals
 class HorseRacing
+  attr_accessor :horses_number
+
+  def initialize
+    @horses_number = gets.to_i
+  end
+
   def find_difference
     strenghts = init_strenghts
     strenghts.each_cons(2).map { |a, b| (b - a).abs }.min
   end
 
   def init_strenghts
-    n = gets.to_i
     strenghts_array = []
-    n.times do
-      pi = gets.to_i
-      strenghts_array << pi
+    horses_number.times do
```

Comment:
Try using [each_with_object](https://ruby-doc.org/core-2.1.0/Enumerator.html#method-i-each_with_object). You can also chain it to `.times`